src/api/rpc_api/state.rs

```diff
@@ -121,65 +150,63 @@ where
 	Runtime: FrameSystemConfig,
 	Params: ExtrinsicParams<Runtime::Index, Runtime::Hash>,
 {
-	fn get_storage_value<V: Decode>(
+	fn get_storage<V: Decode>(
 		&self,
-		storage_prefix: &'static str,
-		storage_key_name: &'static str,
+		pallet: &'static str,
+		storage_item: &'static str,
 		at_block: Option<Runtime::Hash>,
 	) -> Result<Option<V>> {
-		let storagekey = self.metadata().storage_value_key(storage_prefix, storage_key_name)?;
+		let storagekey = self.metadata().storage_value_key(pallet, storage_item)?;
 		info!("storage key is: 0x{}", hex::encode(&storagekey));
-		self.get_storage_by_key_hash(storagekey, at_block)
+		self.get_storage_by_key(storagekey, at_block)
 	}
 
 	fn get_storage_map<K: Encode, V: Decode>(
 		&self,
-		storage_prefix: &'static str,
-		storage_key_name: &'static str,
+		pallet: &'static str,
+		storage_item: &'static str,
 		map_key: K,
 		at_block: Option<Runtime::Hash>,
 	) -> Result<Option<V>> {
-		let storagekey =
-			self.metadata()
-				.storage_map_key::<K>(storage_prefix, storage_key_name, map_key)?;
+		let storagekey = self.metadata().storage_map_key::<K>(pallet, storage_item, map_key)?;
 		info!("storage key is: 0x{}", hex::encode(&storagekey));
-		self.get_storage_by_key_hash(storagekey, at_block)
+		self.get_storage_by_key(storagekey, at_block)
 	}
 
-	fn get_storage_map_key_prefix(
+	fn get_storage_key_prefix(
```

Comment:
Are you sure it's not a map prefix? To my understanding, if there's no map then the storage key can simply be calculated by combining the hashed `pallet` and `storage_item`, no need for another item.

Reply:
changed it back